Changed the behaviour of `~plasmapy.particles.particle_class.CustomParticle.__eq__` to return `False` when used against an other type insted of raising a `TypeError`.
